### ISSUE TITLE
means of suppressing ambiguity warnings

### ISSUE THREAD

**Issue by catawbasam** [JuliaLang/julia#4895]
This issue is carried over from JuliaLang/IJulia.jl#116.

It would be useful at times to have a means of suppressing warnings such as the example below without suppressing errors.

@staticfloat has expressed interest in this issue, as has @StefanKarpinski. /base/util.jl lines 418:446 contains what looks like the seed of a lightweight logging approach along the lines @StefanKarpinski mentioned in the IJulia discussion
 
```
julia> using Winston
Warning: New definition
    |(SynchronousStepCollection,Any) at C:\Users\keithc\.julia\BinDeps\src\BinDe
ps.jl:283
is ambiguous with:
    |(Any,NAtype) at C:\Users\keithc\.julia\DataArrays\src\operators.jl:502.
To fix, define
    |(SynchronousStepCollection,NAtype)
before the new definition.
Warning: New definition
    |(Any,SynchronousStepCollection) at C:\Users\keithc\.julia\BinDeps\src\BinDe
ps.jl:286
is ambiguous with:
    |(NAtype,Any) at C:\Users\keithc\.julia\DataArrays\src\operators.jl:502.
To fix, define
    |(NAtype,SynchronousStepCollection)
before the new definition.
```

**Comment by Sisyphuss** [comment]
This is an old issue, but during the release of v0.4, I would like to add some new thinking about it.

I am plotting in IJulia in using Gadfly. There are tons of deprecation warnings. My plot is hidden in the *middle* of these warnings. For me, it's just inconvenient. But if I would like to send my notebook to someone who knows nothing about Julia, how can I expect that he could find the plot?

**Comment by jakebolewski** [comment]
@Sisyphuss this issue is about ambiguity warnings, not deprecation warnings.

**Comment by wildart** [comment]
@malmaud [Log4jl](https://github.com/wildart/Log4jl.jl)

**Comment by JeffBezanson** [comment]
subsumed by #6190

**Comment by SimonDanisch** [comment]
Not sure since when ambiguities are the only warnings you can get in Julia :P
I think this issue is alive: It's still not easy to present Julia nicely with e.g. Notebooks, since they might be swamped with warnings from packages that are not under your control ;)
Or am I missing some other feature?

**Comment by nalimilan** [comment]
You're likely seeing a different kind of warnings. Would need to see a concrete example to tell (and which Julia version?).

**Comment by SimonDanisch** [comment]
I'm talking about general warnings, like overwritten functions, or any package calling `warn(...)`.
I guess I could just overwrite warn for that purpose, but is that what we want as the solution to this issue? ;)

**Comment by alanedelman** [comment]
I wish I could suppress warnings nearly every time I give a demo,
and almost every time I compute.

**Comment by KristofferC** [comment]
Can't you just redirect them now with the new logging functionality?

**Comment by SimonDanisch** [comment]
Is there new logging functionality? Would that also work on 0.5?

**Comment by KristofferC** [comment]
https://github.com/JuliaLang/julia/pull/16213

**Comment by SimonDanisch** [comment]
Nice. So we can label this as fixed on 0.6!

**Comment by tkelman** [comment]
this issue is about ambiguity warnings which were fixed in 0.5

**Comment by SimonDanisch** [comment]
haha, duh! Guess the discussion got me a bit side tracked... Also this is the first result you get when googling for suppressing warnings! ;)
